File: core/src/main/java/com/sequenceiq/cloudbreak/repository/TemplateRepository.java
package com.sequenceiq.cloudbreak.repository;

import java.util.Set;

import javax.transaction.Transactional;
import javax.transaction.Transactional.TxType;

import org.springframework.data.repository.CrudRepository;

import com.sequenceiq.cloudbreak.domain.Template;
import com.sequenceiq.cloudbreak.domain.Topology;
import com.sequenceiq.cloudbreak.workspace.repository.EntityType;

@EntityType(entityClass = Template.class)
@Transactional(TxType.REQUIRED)
public interface TemplateRepository extends CrudRepository<Template, Long> {

    Set<Template> findByTopology(Topology topology);

}

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/TemplateRepository.kt
package com.sequenceiq.cloudbreak.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param

import com.sequenceiq.cloudbreak.domain.Template

@EntityType(entityClass = Template::class)
interface TemplateRepository : CrudRepository<Template, Long> {

    override fun findOne(@Param("id") id: Long?): Template

    fun findForUser(@Param("user") user: String): Set<Template>

    fun findPublicInAccountForUser(@Param("user") user: String, @Param("account") account: String): Set<Template>

    fun findAllInAccount(@Param("account") account: String): Set<Template>

    fun findOneByName(@Param("name") name: String, @Param("account") account: String): Template

    fun findByNameInAccount(@Param("name") name: String, @Param("account") account: String, @Param("owner") owner: String): Template

    fun findByIdInAccount(@Param("id") id: Long?, @Param("account") account: String): Template

    fun findByNameInUser(@Param("name") name: String, @Param("owner") owner: String): Template

    fun findAllDefaultInAccount(@Param("account") account: String): Set<Template>

    fun findByTopology(@Param("topologyId") topologyId: Long?): Set<Template>

}